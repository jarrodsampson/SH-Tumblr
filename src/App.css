.App {
    margin:auto;
}

.thumbs {
    width:100%;
    padding-bottom:1em;
}

.hidden {
    display:none;
    transition: all 2s;
}