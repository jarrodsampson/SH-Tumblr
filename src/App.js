import React, { Component } from 'react';
import LazyLoad from 'react-lazyload';
import ScrollToTop from 'react-scroll-up';
import './App.css';
import $ from 'jquery';

class App extends Component {

  constructor(props) {
    super(props);
    this.state = {
      tumblrPhotos: [],
      tag: "gaming",
      loading: true
    };
  }

  componentDidMount() {
    this.apiCall();
  }

  apiCall() {
    let apiKey = "QPdARZFwjbG0M0teN9564Apj1kq81wFKkxIb1kurTagPaUEzPY";

    $.getJSON('//api.tumblr.com/v2/tagged?tag=' + this.state.tag + '&api_key=' + apiKey).then(function(res) {
       this.setState({
           tumblrPhotos: res.response,
           loading: false
       });
    }.bind(this));

  }

  loadFunc() {

  }

  render() {
    let photos = this.state.tumblrPhotos;
    console.log(this.state.tumblrPhotos)
    return (
      <div className="App">
        <div className="container">
            <header className="App-header">
              <h1>SkillShare</h1>
              <h2 className="App-title">Searching: {this.state.tag}</h2>
            </header>
            <div className="row">
                <div className={this.state.isLoading ? 'hidden' : ''}>
                    {photos.map(function(photo, i) {
                        let url = "";

                        if (photo.photos)
                        {
                            url = photo.photos[0].original_size.url;
                        }

                        return <div className="col-md-12" key={i}>
                            <LazyLoad height={200}>
                                <img className="thumbs" src={url || "http://via.placeholder.com/350x150"} alt={photo.blog_name} />
                            </LazyLoad>
                        </div>;
                    })}
                </div>
            </div>
        </div>

        <ScrollToTop showUnder={160}>
          <span>UP</span>
        </ScrollToTop>

      </div>
    );
  }
}

export default App;
