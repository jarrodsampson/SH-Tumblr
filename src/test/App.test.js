import React from 'react';
import ReactDOM from 'react-dom';
import App from '../App';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';


describe('Test Suite', () => {

    beforeAll(() => {});

    beforeEach(() => {});

    it('renders without crashing', () => {
      let div = document.createElement('div');
      ReactDOM.render(<App />, div);
      ReactDOM.unmountComponentAtNode(div);
    });

    it('renders correctly', () => {
      let tree = renderer
        .create(<App />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it('renders header correctly', () => {
      let wrapper = shallow(<App />);
      expect(wrapper.find('.App .App-header h1').children().text()).toEqual("SkillShare");
    });

    afterAll(() => {});

    afterEach(() => {});

});