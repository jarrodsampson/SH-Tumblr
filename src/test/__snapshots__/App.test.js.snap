// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Suite renders correctly 1`] = `
<div
  className="App"
>
  <div
    className="container"
  >
    <header
      className="App-header"
    >
      <h1>
        SkillShare
      </h1>
      <h2
        className="App-title"
      >
        Searching: 
        travel
      </h2>
    </header>
    <div
      className="row"
    >
      <div
        className=""
      />
    </div>
  </div>
</div>
`;
